AppCenterDataStorage/AppCenterDataStorageTests/MSDataStoreTests.m

```diff
@@ -243,6 +243,63 @@ - (void)testCreateWhenDataModuleDisabled {
   XCTAssertEqualObjects(actualDocumentWrapper.documentId, kMSDocumentIdTest);
 }
 
+- (void)testReplaceWithPartitionGoldenPath {
+  
+  // If
+  XCTestExpectation *expectation = [self expectationWithDescription:@"Replace with partition completed"];
+  id<MSSerializableDocument> mockSerializableDocument = [MSFakeSerializableDocument new];
+  __block BOOL completionHandlerCalled = NO;
+  __block MSDocumentWrapper *actualDocumentWrapper;
+  MSTokenResult *testToken = [self mockTokenFetchingWithError:nil];
+  
+  // Mock CosmosDB requests.
+  NSData *testCosmosDbResponse = [self jsonFixture:@"validTestDocument"];
+  OCMStub([self.cosmosDbMock performCosmosDbAsyncOperationWithHttpClient:OCMOCK_ANY
+                                                             tokenResult:testToken
+                                                              documentId:kMSDocumentIdTest
+                                                              httpMethod:kMSHttpMethodPost
+                                                                document:mockSerializableDocument
+                                                       additionalHeaders:OCMOCK_ANY
+                                                       additionalUrlPath:OCMOCK_ANY
+                                                       completionHandler:OCMOCK_ANY])
+      .andDo(^(NSInvocation *invocation) {
+        MSHttpRequestCompletionHandler cosmosdbOperationCallback;
+        [invocation getArgument:&cosmosdbOperationCallback atIndex:9];
+        cosmosdbOperationCallback(testCosmosDbResponse, [self generateResponseWithStatusCode:MSHTTPCodesNo200OK], nil);
+      });
+
+  // When
+  [MSDataStore createWithPartition:kMSPartitionTest
+                        documentId:kMSDocumentIdTest
+                          document:mockSerializableDocument
+                 completionHandler:^(MSDocumentWrapper *data) {
+                   completionHandlerCalled = YES;
+                   actualDocumentWrapper = data;
+                   [expectation fulfill];
+                 }];
+  id<MSSerializableDocument> replaceMockSerializableDocument = [MSFakeSerializableDocument new];
+  [MSDataStore replaceWithPartition:kMSPartitionTest
```

Comment:
I don't see a test case covering the replace API which takes `writeOptions:` as parameter. Please consider adding a test for it.

Reply:
It is called implecitly with nil options. But I can add a test case for writeOptions, too :)

Reply:
Let's not block the PR on that, the read/write stuff is being tested at the data proxy level already.

Thanks for catching that Anna!

Reply:
We decided to unblock this.